streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransform.java

```diff
@@ -59,6 +59,7 @@ public void process(K1 key, V1 value) {
                 context().forward(pair.key, pair.value);
         }
 
+        @SuppressWarnings("deprecation")
```

Comment:
Its suppressing the warning for calling deprecated `punctuate` method that we need to keep for backward compatibility. (actually not part of this KIP but a side cleanup)